msal/authority.py

```diff
@@ -71,6 +71,14 @@ def __init__(self, authority_url, http_client, validate_authority=True):
         if isinstance(authority_url, AuthorityBuilder):
             authority_url = str(authority_url)
         authority, self.instance, tenant = canonicalize(authority_url)
+
+        # extract authority port
```

Comment:
Can we have some introduction in this PR's description, to describe the scenario that you are trying to support?